src/app/product-add/product-add.component.ts

```diff
@@ -12,38 +15,32 @@ export class ProductAddComponent implements OnInit {
     name : new FormControl('', Validators.required),
     category: new FormControl('', Validators.required),
     image: new FormControl('', Validators.required),
-    price: new FormControl('', [Validators.required, Validators.min(1), Validators.pattern("^[1-9]*$")]),
+    price: new FormControl('', [Validators.required, Validators.min(1), Validators.pattern("^[0-9]*$")]),
     description: new FormControl('', Validators.required)
   });
 
-  constructor(private productService: ProductsService) { }
+  constructor(private location: Location, private store: Store<AppState>) { }
 
   ngOnInit(): void {
   }
 
-  refreshForm() {
-    this.addForm.setValue({
-      name: "",
-      category: "",
-      image: "",
-      price: "",
-      description: ""
-    });
+  onSubmit() : void {
+    if ( this.addForm.value.name && this.addForm.value.price && this.addForm.value.category && this.addForm.value.image && this.addForm.value.description) {
+      const newProduct = {
+        id: Math.random(),
+        name: this.addForm.value.name,
+        category: this.addForm.value.category,
+        image: this.addForm.value.image,
+        price: parseInt(this.addForm.value.price),
+        description: this.addForm.value.description
+      }
+      this.store.dispatch(addProduct({product: newProduct}));
+      this.addForm.reset();
```

Comment:
There is no need to reset the form if you redirect to another page. The form will be re-created if this page is accessed again.